code/modules/reagents/Chemistry-Holder.dm

```diff
@@ -345,7 +345,8 @@ trans_to_atmos(var/datum/gas_mixture/target, var/amount=1, var/multiplier=1, var
 		var/datum/reagent/R = A
 		if(M && R)
 			R.on_mob_life(M, alien)
-			R.metabolize(M)
+			if(R)
```

Comment:
What necessitates this check?

Reply:
Because `on_mob_life()` for chems like charcoal will delete themselves, and that was causing runtimes.